### PR TITLE
Add the VOiCES recipe for ASR

### PR DESCRIPTION
**Pull request by Emrys365**
## What?

This PR adds a new ASR recipe based on the [VOiCES](https://iqtlabs.github.io/voices/) corpus.

* License: Creative Commons BY 4.0
* Audio was recorded in two furnished rooms with background noise played in conjunction with foreground speech selected from the LibriSpeech corpus (300 speakers).
* 12 distributed microphones, 120 hours of audio per microphone
* A dataset for ASR and speaker verification

## Why?

This can be a good candidate for evaluating far-field ASR performance with distributed microphones.

## See also

[Voices Obscured in Complex Environmental Settings (VOICES) corpus](https://arxiv.org/pdf/1804.05053.pdf)

### CODE REVIEW

File: egs2/voices/asr1/local/data.sh [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        done
+
+        utils/utt2spk_to_spk2utt.pl data/${train_set}${suffix}_tmp/utt2spk > data/${train_set}${suffix}_tmp/spk2utt
+        utils/utt2spk_to_spk2utt.pl data/${test_set}${suffix}/utt2spk > data/${test_set}${suffix}/spk2utt
+
+        utils/filter_scp.pl data/${train_set}${suffix}_tmp/wav.scp data/local/voices_text${suffix} | sort > data/${train_set}${suffix}_tmp/text
+        utils/filter_scp.pl data/${test_set}${suffix}/wav.scp data/local/voices_text${suffix} | sort > data/${test_set}${suffix}/text
+
+        # split ${train_set}${suffix}_tmp set to ${train_set} and ${valid_set}
+        head -n 10 data/${train_set}${suffix}_tmp/spk2utt > data/${train_set}${suffix}_tmp/dev_spk2utt
```
Comment (Emrys365) [inline egs2/voices/asr1/local/data.sh:171]: Note that I always take utterances from the first 10 speakers as the dev set. I'm not sure whether this is a reasonable setting.
There is no official split for training and development sets.

File: egs2/voices/asr1/conf/train_asr_conformer.yaml
Comment (Emrys365) [inline egs2/voices/asr1/conf/train_asr_conformer.yaml:1]: The config files are basically copied from the Librispeech recipe as the VOiCES data are generated based on Librispeech.

File: egs2/voices/asr1/local/data.sh (hunk 1 above)
Comment (Emrys365) [reply egs2/voices/asr1/local/data.sh:171]: This results in the following amount of samples in each subset:
```bash
egs2/voices/asr1/data/
├── full_dev
│   ├── spk2utt (10 speakers)
│   ├── text
│   ├── utt2spk
│   └── wav.scp (33924 utterances)
├── full_dev_multich
│   ├── spk2utt (10 speakers)
│   ├── text
│   ├── utt2spk
│   └── wav.scp (6336 utterances)
├── full_test
│   ├── spk2utt (100 speakers)
│   ├── text
│   ├── utt2spk
│   └── wav.scp (339240 utterances)
├── full_test_multich
│   ├── spk2utt (100 speakers)
│   ├── text
│   ├── utt2spk
│   └── wav.scp (63360 utterances)
├── full_train
│   ├── spk2utt (190 speakers)
│   ├── text
│   ├── utt2spk
│   └── wav.scp (629907 utterances)
└── full_train_multich
    ├── spk2utt (190 speakers)
    ├── text
    ├── utt2spk
    └── wav.scp (117648 utterances)
```

File: egs2/voices/asr1/conf/decode_asr.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+beam_size: 60
```
Comment (ftshijt) [inline egs2/voices/asr1/conf/decode_asr.yaml:1]: did not usually see 60 beam size, any specific reasons?

**Review by ftshijt — COMMENTED** [review]
Thanks for the PR and LGTM! Just a minor question:

File: egs2/voices/asr1/conf/decode_asr.yaml (hunk 2 above)
Comment (Emrys365) [reply egs2/voices/asr1/conf/decode_asr.yaml:1]: I simply borrow this config from egs2/librispeech/asr1. The beam size seems to be updated by @sw005320.

Comment (sw005320) [reply egs2/voices/asr1/conf/decode_asr.yaml:1]: 60 is too large.
20 is fine.

**Comment by Emrys365** [comment]
Can we merge this PR?

**Comment by sw005320** [comment]
Thanks!
Please add README.md after you got some results.

**Pull request merged**